juju/juju
# K8s labels dev

### Checklist


----

## Description of change

Changes the labels used by Juju for Kubernetes.

Juju now adopts more idiomatic way of adding Kubernetes labels as per k8s docs https://kubernetes.io/docs/concepts/overview/working-with-objects/common-labels/

We are adopting k8s standard labels and also adding juju.is labels such as model.juju.is/name

This change will not retroactively change labels for existing models

## QA steps

Need to perform a lot of bootstrapping and upgrading older controllers.

- We are wanting to make sure that model's pre this change continue to use the older style of kubernetes labeling.
- Need to make sure that newly created models use the new style of labels
- All created objects are correctly labels include a managed-by: juju label

```sh
#Run Unit Tests
cd caas/kubernetes/provider
go test .
```

## Documentation changes

- Needs a discourse post explaining the change

## Bug reference

https://bugs.launchpad.net/juju/+bug/1888513
